static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+        (x) =>
+          x.filter === params.filter &&
+          x.instrument_name === params.instrument &&
+          x.mjd >= mjdmin &&
+          x.mjd <= mjdmax
+      );
+      const times = data.map((x) => x.mjd);
+      const mag = data.map((x) => x.mag);
+      const title = `P=${bestp?.toFixed(8)} d (${params.instrument}: ${
+        params.filter
+      })`;
+      plotphased(times, mag, bestp, title);
+    }
+  }, [id, photometry, bestp, run, dispatch]);
+
+  const componentRef = useRef();
+
+  const initialFormState = {
+    ...params,
+  };
+
+  const onSubmit = () => {
+    const formData = {
+      ...initialFormState,
+      ...getValues(),
+    };
+    setPlotted(false);
+    setRun(false);
+    setParams(formData);
+  };
+
+  const rules = { required: true, min: 1, max: 25, type: "number", step: 1 };
+
+  return (
+    <>
+      <div>
+        <div>
+          <Typography variant="h5" gutterBottom>
+            Interactive Periodogram for{" "}
+            <Link to={`/source/${id}`} role="link">
+              {id}
+            </Link>
+          </Typography>
+        </div>
+        <Grid
+          container
+          direction="row"
+          justify="flex-start"
+          alignItems="flex-start"
+          spacing={1}
+        >
+          <Grid item xs={10}>
+            <Card>
+              <CardContent ref={componentRef}>
+                <div>
+                  {photometry ? (
+                    <div ref={dataplotRef} className={classes.dygraphChart} />
+                  ) : (
+                    <div>{placeholder}</div>
+                  )}
+                </div>
+                <div>
+                  <p />
+                  <div ref={glsplotRef} className={classes.dygraphChart} />
+                </div>
+                <div>
+                  <p />
+                  <div ref={phaseplotRef} className={classes.dygraphChart} />
+                </div>
+              </CardContent>
+            </Card>
+          </Grid>
+          <Grid item xs={2} xm={2}>
+            <Card>
+              <CardContent className={classes.items}>
+                <div>
+                  <form onSubmit={handleSubmit(onSubmit)}>
+                    <Grid
+                      container
+                      direction="column"
+                      justify="space-evenly"
+                      alignItems="flex-start"
+                      spacing={2}
+                    >
+                      <Grid item xs={12}>
+                        <FormControl>
+                          <InputLabel
+                            className={classes.items}
+                            id="InstrumentSourceSelect"
+                          >
+                            Instrument
+                          </InputLabel>
+                          <p />
+                          <Controller
+                            as={Select}
+                            labelid="InstrumentSourceSelectLabel"
```

Comment:
`labelid` should be `labelId`, and the ID appears to not match the ID of the label above

Reply:
Good catch. Fixed.